lib/pure/random.nim

```diff
@@ -158,10 +158,14 @@ proc rand*[T](x: HSlice[T, T]): T =
   result = rand(state, x)
 
 proc rand*[T](r: var Rand; a: openArray[T]): T {.deprecated.} =
-  ## returns a random element from the openarray `a`.
+  ## Returns a random element from the openarray `a`.
   ## **Deprecated since v0.20.0:** use ``sample`` instead.
   result = a[rand(r, a.low..a.high)]
 
+proc rand*[T: SomeInteger](t: typedesc[T]): T =
```

Comment:
Perhaps this should use `Ordinal|uint|uint64` instead of `SomeInterger`

Reply:
Or perhaps I should not. Are there any reasons why I should do that? I can only come up with reasons why I should not do that.

Reply:
After thinking about it some more I see why this implementation wouldn't work for arbitrary ordinals...